### ISSUE TITLE
after fix Issue #63, iconv(3) can't detect incomplete input or shift state.

### ISSUE THREAD

**Issue by ikazont** [ikazont/netbsd#72]
**[Original report](https://bitbucket.org/tnozaki/netbsd/issues/72) by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

try following:

```
#!sh

$ printf "\xA1" | iconv -f eucJP -t UTF-8

$ printf "\xA1" | /usr/pkg/bin/iconv -f eucJP -t UTF-8
/usr/pkg/bin/iconv: (stdin):1:0: incomplete character or shift sequence
```

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

my idea is:

* all encoding module should implement _citrus_stdenc_mbtocs_state_reset() and if incomplete chars in mbstate_t, throw EINVAL.
* iconv_std checks mbstate_t by _citrus_stdenc_get_state_desc() before call _citrus_stdenc_mbtocs_state_reset().

**Comment by ikazont** [comment]
**Original comment by Takehiko NOZAKI (Bitbucket: [tnozaki](https://bitbucket.org/tnozaki), GitHub: [ikazont](https://github.com/ikazont)).**

----------------------------------------

BUGFIX: Issue #72 -- after fix Issue #63, iconv(3) can't detect incomplete input or shift state.
